pkg/cmd/server/server.go

```diff
@@ -615,87 +654,134 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
+			s.logger,
+			s.sharedInformerFactory.Velero().V1().ResticRepositories(),
+			s.veleroClient.VeleroV1(),
+			s.sharedInformerFactory.Velero().V1().BackupStorageLocations(),
+			s.resticManager,
+		)
 
-	serverStatusRequestController := controller.NewServerStatusRequestController(
-		s.logger,
-		s.veleroClient.VeleroV1(),
-		s.sharedInformerFactory.Velero().V1().ServerStatusRequests(),
-	)
-	wg.Add(1)
-	go func() {
-		serverStatusRequestController.Run(ctx, 1)
-		wg.Done()
-	}()
+		return controllerRunInfo{
+			controller: resticRepoController,
+			numWorkers: defaultControllerWorkers,
+		}
+	}
+
+	downloadrequestControllerRunInfo := func() controllerRunInfo {
+		downloadRequestController := controller.NewDownloadRequestController(
+			s.veleroClient.VeleroV1(),
+			s.sharedInformerFactory.Velero().V1().DownloadRequests(),
+			s.sharedInformerFactory.Velero().V1().Restores(),
+			s.sharedInformerFactory.Velero().V1().BackupStorageLocations(),
+			s.sharedInformerFactory.Velero().V1().Backups(),
+			newPluginManager,
+			s.logger,
+		)
+
+		return controllerRunInfo{
+			controller: downloadRequestController,
+			numWorkers: defaultControllerWorkers,
+		}
+	}
+
+	serverStatusRequestControllerRunInfo := func() controllerRunInfo {
+		serverStatusRequestController := controller.NewServerStatusRequestController(
+			s.logger,
+			s.veleroClient.VeleroV1(),
+			s.sharedInformerFactory.Velero().V1().ServerStatusRequests(),
+		)
+
+		return controllerRunInfo{
+			controller: serverStatusRequestController,
+			numWorkers: defaultControllerWorkers,
+		}
+	}
+
+	enabledControllers := map[string]func() controllerRunInfo{
+		BackupSyncControllerKey:          backupSyncControllerRunInfo,
+		BackupControllerKey:              backupControllerRunInfo,
+		ScheduleControllerKey:            scheduleControllerRunInfo,
+		GcControllerKey:                  gcControllerRunInfo,
+		BackupDeletionControllerKey:      deletionControllerRunInfo,
+		RestoreControllerKey:             restoreControllerRunInfo,
+		ResticRepoControllerKey:          resticRepoControllerRunInfo,
+		DownloadRequestControllerKey:     downloadrequestControllerRunInfo,
+		ServerStatusRequestControllerKey: serverStatusRequestControllerRunInfo,
+	}
+
+	if s.config.restoreOnly {
+		s.logger.Info("Restore only mode - not starting the backup, schedule, delete-backup, or GC controllers")
+		s.config.disabledControllers = append(s.config.disabledControllers,
+			BackupControllerKey,
+			ScheduleControllerKey,
+			GcControllerKey,
+			BackupDeletionControllerKey,
+		)
+	}
+
+	// remove disabled controllers
+	for _, controllerName := range s.config.disabledControllers {
+		if _, ok := enabledControllers[controllerName]; ok {
+			s.logger.Infof("Disabling controller: %s", controllerName)
+			delete(enabledControllers, controllerName)
+		} else {
+			s.logger.Fatalf("Invalid value for --disable-controllers flag provided: %s", controllerName)
+		}
```

Comment:
Could we enumerate the valid values in the error? I know it's in the help message, but this removes a little friction for users.

Reply:
Added!